# HG changeset patch
# User Hartmut Gimpel <hg_code@gmx.de>
# Date 1477257362 -7200
#      Sun Oct 23 23:16:02 2016 +0200
# Branch stable
# Node ID f88de8f2725937890f87957c619ffad99fc0a222
# Parent  c024fb1031148ec099e256b672fed76fcef61e13
fix text encoding in qt4 under Linux (bug #49118)

* octave-gui.cc (gui_application::execute): For Qt4, restore call to
QTextCodec::setCodecForCStrings that was removed in changeset
2fd4f1a3f4a8.

diff --git a/libgui/src/octave-gui.cc b/libgui/src/octave-gui.cc
--- a/libgui/src/octave-gui.cc
+++ b/libgui/src/octave-gui.cc
@@ -154,16 +154,20 @@ namespace octave
     QApplication qt_app (m_argc, m_argv);
     QTranslator gui_tr, qt_tr, qsci_tr;
 
     // Set the codec for all strings (before wizard or any GUI object)
 #if ! defined (Q_OS_WIN32)
     QTextCodec::setCodecForLocale (QTextCodec::codecForName ("UTF-8"));
 #endif
 
+#if defined (HAVE_QT4)
+    QTextCodec::setCodecForCStrings (QTextCodec::codecForName ("UTF-8"));
+#endif
+
     // set windows style for windows
 #if defined (Q_OS_WIN32)
     qt_app.setStyle(QStyleFactory::create("Windows"));
 #endif
 
     bool start_gui = start_gui_p ();
 
     // Show welcome wizard if this is the first run.
